state/api/src/main/java/com/expediagroup/streamplatform/streamregistry/state/EntityView.java

```diff
@@ -63,4 +57,25 @@
    * @return a stream containing all entities of the given key type.
    */
   <K extends Key<S>, S extends Specification> Stream<Entity<K, S>> all(Class<K> keyClass);
+
+  /**
+   * Returns a {@link Map} containing all keys of the given {@link Key} type which have been deleted but not
+   * purged ({@link #purgeDeleted(Key)}) mapped to the deleted Entity (if known)
+   *
+   * @param keyClass the key class of an entity type.
+   * @param <K>      the key type.
+   * @param <S>      the specification type.
+   * @return a map containing all deleted keys of a give type to the key if known
+   */
+  <K extends Key<S>, S extends Specification> Map<K, Optional<Entity<K, S>>> allDeleted(Class<K> keyClass);
```

Comment:
Yeah, I'll try and articulate the case just so I don't forget 😄 

Say we have the following on a backing kafka topic:

| Key | Value |
|---|---|
| `K1` | `V1` |

After bootstrapping the backing Map contains `K1` -> `(K2, deleted=false)`.

We then issue a delete:

| Key | Value |
|---|---|
| `K1` | `V1` |
| `K1` | `null` |

After bootstrapping the backing Map contains `K1` -> `(K2, deleted=true)`.

Sometime in the future, compaction will run. But this does not cause tombstones to be deleted right away. (This is by design in Kafka, in order to give consumers time to register the delete). So we end up with:

| Key | Value |
|---|---|
| `K1` | `null` |

Now we no longer have the original entity. So the backing map looks like:

`K1` -> `(null, deleted=true)`.

I consumed from a Kafka topic in usage and could confirm this behaviour.

Reply:
The above being said, you are correct that the contract needs more explanation in these java docs 😄

Reply:
Done, though I've not included any references to the specific kafka case since this class doesn't have requirements on the underlying storage mechanism.